### PR TITLE
[@types/stripe] Added config param to constructor

### PR DESCRIPTION
**Pull request by allouis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

~^ Running `npm run lint stripe` gives back an error like:~
```
The 'no-unnecessary-generics' rule threw an error in '/usr/src/js/DefinitelyTyped/types/stripe/index.d.ts':
TypeError: sig.parameters is not iterable
```

~:thinking: It doesn't seem like this is related to this change - but I'm not sure how to fix it :|~

Looks like it passes on CI :relaxed: 

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/stripe/stripe-node#initialize-with-config-object
- [ ] ~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~
- [ ] ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~

### CODE REVIEW

**Comment by acomagu** [comment]
Oh, dups with mine #40841...

**Comment by acomagu** [comment]
Your pull is better than mine because yours keeps the line `constructor(apiKey: string, version?: string);` which is fit to the real implementation.
So I closed mine. Hope to merge soon!(I'm beginner to contributing DefinitlyTyped so I can't review and approve, sorry)

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -63,6 +63,16 @@ interface ResponseEvent {
     request_end_time: number;
 }
 
+interface StripeConfig {
+    apiVersion: string;
```
Comment (pushplay) [inline types/stripe/index.d.ts]: From the doc it looks like all the properties here should be marked optional.  As is all properties have to be specified.

`apiVersion` and `httpAgent` also allow null so their types should be `string | null` and `Agent | null` respectively.

**Review by pushplay — CHANGES_REQUESTED** [review]
Thanks for improving the tests.

**Comment by allouis** [comment]
Thanks for the review! I've updated the types and added another test :relaxed:

**Review by pushplay — APPROVED** [review]
Thanks for the updates.

**Comment by orta** [comment]
Looks like additions only to me, thanks!

**Pull request merged**